client/engine/messageservice/test-messageservice.go

```diff
@@ -0,0 +1,75 @@
+package messageservice
+
+import (
+	"fmt"
+
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/types"
+)
+
+// TestMessageService is an implementaion of the MessageService interface
+// for use in with-peers style test environments.
+//
+// It allows for individual nitro-clients / engines to:
+//  1. be instantiated together via test setup data
+//  2. "connect" with one another via gochans
+//  3. run independently in information-silo goroutines, while
+//     communicating on the simulated network
+type TestMessageService struct {
+	address types.Address
+
+	toPeers map[types.Address]chan<- protocols.Message
+	out     chan protocols.Message
+
+	in chan protocols.Message
+}
+
+func (t TestMessageService) Run() {
+	go t.routeOutgoing()
+}
+
+func (t TestMessageService) GetReceiveChan() chan protocols.Message {
+	return t.in
+}
+
+func (t TestMessageService) GetSendChan() chan<- protocols.Message {
+	return t.out
+}
+
+func (t TestMessageService) Send(message protocols.Message) {
+	t.out <- message
+}
+
+// Connect creates a gochan for message service t to communicate with
+// the given peer. This connection is one-way.
```

Comment:
Nit:
```suggestion
// Connect creates a gochan for message service t to send messages to the given peer.
```

"send" is better than "communicate one way" (IMO).

Reply:
Meta comment: the receiver variable `t` doesn't make a ton of sense when a dev hovers over the method on a test message service instance they are working with (they may not have named it `t`). So does `t` belong in the documentation here?